src/CacheDirectoryFactory/CacheDirectoryFactory.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crunz\CacheDirectoryFactory;
+
+use Crunz\Path\Path;
+
+final class CacheDirectoryFactory
+{
+    public const CRUNZ_BASE_CACHE_PATH = 'CRUNZ_BASE_CACHE_PATH';
+
+    public function generate(): Path
+    {
+        $baseCacheDirectory = ($basePath = \getenv(self::CRUNZ_BASE_CACHE_PATH, true)) !== false
```

Comment:
Let's move  `$basePath` to separate line, also what about values other than `false`, like `''` (empty string) or `null`?

Reply:
[According to docs](https://www.php.net/manual/en/function.getenv.php) `array` is returned only if provided `name` is `null`, so not our case. `false` means that variable does not exist (so no exception here), but empty string should throw an exception, because it looks like user is trying to use that feature, but with wrong value

Reply:
I added an ltrim to the path with the assumption that an absolute path does not start with whitespaces and to make it easier to catch the empty string case.